test/deployframework/local.go

```diff
@@ -0,0 +1,206 @@
+		"REPORTING_OPERATOR_PPROF_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, pprofListenPort),
+		"REPORTING_OPERATOR_PROMETHEUS_DATASOURCE_MAX_IMPORT_BACKFILL_DURATION=15m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_INTERVAL=30s",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_CHUNK_SIZE=5m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_INTERVAL=5m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_STEP_SIZE=60s",
+		"REPORTING_OPERATOR_TLS_DIR=" + lc.ReportingOperatorTLSPath,
+	}
+
+	// TODO: add this to LocalCtx?
+	scriptPath, err := filepath.Abs("../../hack/run-reporting-operator-local.sh")
+	if err != nil {
+		return fmt.Errorf("failed to get the absolute path to the hack/run-reporting-operator-local.sh script: %v", err)
+	}
+
+	cmd := exec.Command(scriptPath)
+
+	cmd.Env = append(os.Environ(), envVarArr...)
+
+	lc.Logger.WithFields(logrus.Fields{
+		"command":     cmd.Args,
+		"environment": envVarArr,
+	}).Infof("Running the reporting-operator locally")
+
+	logFile, err := ioutil.TempFile(lc.BasePath, reportingOperatorLogName)
+	if err != nil {
+		return fmt.Errorf("failed to create the local reporting-operator log file: %v", err)
+	}
+
+	lc.Logger.Infof("Storing the hack/run-reporting-operator-local.sh logs to the '%s' path", logFile.Name())
+
+	cmd.Stdout = logFile
+	cmd.Stderr = logFile
+
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("failed to run the reporting-operator locally: %v", err)
+	}
+
+	lc.ReportingOperatorPID = cmd.Process.Pid
+
+	return nil
+}
+
+// WaitForReportingOperator runs a GET request for the reporting-operator healthy endpoint
+func (lc *LocalCtx) WaitForReportingOperator(reportingAPIURL string) error {
+	u, err := url.Parse(reportingAPIURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse the '%s' reporting api URL: %v", reportingAPIURL, err)
+	}
+
+	lc.Logger.WithFields(logrus.Fields{
+		"GET": u,
+	}).Infof("Waiting for the reporting-operator to become ready")
+
+	err = wait.Poll(10*time.Second, 5*time.Minute, func() (done bool, err error) {
+		resp, err := http.Get(u.String())
+		if err != nil {
+			return false, nil
+		}
+		defer resp.Body.Close()
+
+		return resp.StatusCode == http.StatusOK, nil
+	})
+	if err != nil {
+		return fmt.Errorf("timed-out while waiting for the reporting-operator to become healthy")
+	}
+
+	lc.Logger.Infof("The reporting-operator is healthy")
+
+	return nil
+}
+
+// CleanupLocal removes all resources that were created while running e2e locally
+func (lc *LocalCtx) CleanupLocal() error {
+	var errArr []string
+
+	lc.Logger.Infof("Removing the '%s' reporting-operator TLS directory", lc.ReportingOperatorTLSPath)
+	err := os.RemoveAll(lc.ReportingOperatorTLSPath)
+	if err != nil {
+		errArr = append(errArr, fmt.Sprintf("failed to remove the directory containing the reporting-operator TLS data: %v", err))
+	}
+
+	err = cleanupPID(lc.Logger, lc.MeteringOperatorPID, lc.ReportingOperatorPID)
```

Comment:
We should probably only be doing this after `docker rm -f` which should stop the docker container also.